src/common/util.js

```diff
@@ -401,7 +401,7 @@ var Cookies = {
       return loginids.map(function(_id) { return _id.id }).indexOf(id.id) === -1;
     });
 
-    return oauth_loginids ? oauth_loginids : loginids
+    return oauth_loginids.length > 0 ? oauth_loginids : loginids
```

Comment:
@arnabk The statement above, which initializes the `oauth_loginids`, atleast returns an empty array, that's why I didn't put the extra check.